tests/ClientResolverTest.php

```diff
@@ -1519,4 +1520,191 @@ public function testResolveFromEnvIniUseAwsSharedFiles(): void
+        $cases = [
+            'provided_at_client_construction' => [
+                'is_expected' => true,
+                'expected' => [
+                    'smithy.api#httpBearerAuth',
+                    'smithy.api#noAuth',
+                ],
+                'args' => [
+                    'auth_scheme_preference' => [
+                        'smithy.api#httpBearerAuth',
+                        'smithy.api#noAuth',
+                    ]
+                ],
+            ],
+            'provided_auth_scheme_preference_from_env_1' => [
+                'is_expected' => true,
+                'expected' => [
+                    'aws.auth#sigv4a',
+                    'smithy.api#noAuth',
+                ],
+                'args' => [],
+                'fns' => [
+                    'build_source' => function () {
+                        putenv(
+                            'AWS_AUTH_SCHEME_PREFERENCE=aws.auth#sigv4a,smithy.api#noAuth',
+                        );
+                    },
+                    'on_completion' => function () {
+                        putenv('AWS_AUTH_SCHEME_PREFERENCE=');
+                    }
+                ]
+            ],
+            'provided_auth_scheme_preference_from_env_ignore_spaces' => [
+                'is_expected' => true,
+                'expected' => [
+                    'aws.auth#sigv4a',
+                    'smithy.api#noAuth',
+                ],
+                'args' => [],
+                'fns' => [
+                    'build_source' => function () {
+                        putenv(
+                            'AWS_AUTH_SCHEME_PREFERENCE=aws.auth#sigv4a,      smithy.api#noAuth',
+                        );
+                    },
+                    'on_completion' => function () {
+                        putenv('AWS_AUTH_SCHEME_PREFERENCE=');
+                    }
+                ]
+            ],
+            'provided_auth_scheme_preference_from_ini' => [
+                'is_expected' => true,
+                'expected' => [
+                    'smithy.api#noAuth',
+                    'aws.auth#sigv4'
+                ],
+                'args' => [],
+                'fns' => [
+                    'build_source' => function () use ($tempDir, $tempConfigFile) {
+                        if (!is_dir($tempDir)) {
+                            mkdir($tempDir, 0777, true);
+                        }
+
+                        $configContent = <<<EOF
+[default]
+auth_scheme_preference=smithy.api#noAuth, aws.auth#sigv4 
+EOF;
+
+                        file_put_contents($tempConfigFile, $configContent);
+                        putenv("AWS_CONFIG_FILE=$tempConfigFile");
+                    },
+                    'on_completion' => function () use ($tempDir, $tempConfigFile) {
+                        putenv('AWS_CONFIG_FILE=');
+                        unlink($tempConfigFile);
+                        rmdir($tempDir);
```

Comment:
This is not needed- it will cause failures.  Unlinking is good enough

Reply:
So you mean we don't need to remove the directory?
Which or why it will cause failures?

Reply:
I kept using rmdir since I think is important to clean up any resources we use in a test. I am unliking/deleting the only file I am putting in there, so I don't think it should fail.